README.md

```diff
@@ -362,18 +364,87 @@ load test_helper.bash
 will source the script `test/test_helper.bash` in your test file (limitations
 apply, see below). This can be useful for sharing functions to set up your
 environment or load fixtures. `load` delegates to Bash's `source` command after
-resolving relative paths.
+resolving paths.
+
+If `load` encounters errors - e.g. because the targeted source file
+errored - it will print a message with the failing library and Bats
+exits.
+
+To allow to use `load` in conditions `load_safe` has been added.
+`load_safe` prints a message and returns `1` if a source file cannot be
+loaded instead of exiting Bats.
+Aside from that `load_safe` acts exactly like `load`.
 
 As pointed out by @iatrou in https://www.tldp.org/LDP/abs/html/declareref.html,
 using the `declare` builtin restricts scope of a variable. Thus, since actual
 `source`-ing is performed in context of the `load` function, `declare`d symbols
 will _not_ be made available to callers of `load`.
 
+#### `load` argument resolution
+
+`load` supports the following arguments:
+- absolute paths
+- relative paths (to the current test file)
+- library names
+
+`load` tries to determine the `library load path` for an argument
+- a `library load path` is any of the following in order of precedence:
+
+- `argument.bash`
+- `argument` (and is a file)
+- `argument/load.bash`
+- `argument/load`
+- `argument` (and is a directory)
+
 > For backwards compatibility `load` first searches for a file ending in
 > `.bash` (e.g. `load test_helper` searches for `test_helper.bash` before
 > it looks for `test_helper`). This behaviour is deprecated and subject to
 > change, please use exact filenames instead.
 
+If `argument` is an absolute path `load` tries to determine the load
+path directly.
+
+If `argument` is a relative path or a name `load` looks for a matching
+path in the directory of the current test and in all members of
+`BATS_LIB_PATH`.
+
+For example if your `BATS_LIB_PATH` is set to
+`~/.bats/libs:/var/lib/bats-libs`, then `load test_helper` in
+`test/foo.bats` would look in the following paths:
+
+- `test/test_helper.bash`
+- `test/test_helper`
+- `test/test_helper/load.bash`
+- `test/test_helper/load`
+- `test/test_helper` (directory)
+- `~/.bats/libs/test_helper.bash`
+- `~/.bats/libs/test_helper`
+- `~/.bats/libs/test_helper/load.bash`
+- `~/.bats/libs/test_helper/load`
+- `~/.bats/libs/test_helper` (directory)
+- `/var/lib/bats/test_helper.bash`
+- `/var/lib/bats/test_helper`
+- `/var/lib/bats/test_helper/load.bash`
+- `/var/lib/bats/test_helper/load`
+- `/var/lib/bats/test_helper` (directory)
+
+If the found load path is a file it is sourced.
+If the found load path is a directory all files ending `.bash` inside of
+the directory are sourced in alphanumerical order.
+
+If `source` returns with an error on any of the source files `load`
+prints a message and Bats exits.
+
+__Note:__ `BATS_LIB_PATH` defaults to `$HOME/.bats/lib:/usr/lib/bats` if
```

Comment:
The install path of bats is actually `$PREFIX/lib/bats-core` I think we should harmonize that.

Reply:
I decided to keep the internal lib/bats-core separate from the helper libs, so we will stay with `/usr/lib/bats`